### ISSUE TITLE
Underscores in code block captions breaks output

### ISSUE THREAD

**Issue by Xenopathic** [lierdakil/pandoc-crossref#170]
With an underscore in code block captions, e.g.:

``````
``` {.c #lst:foobar caption="foo_bar"}
void foobar(int);
```
``````

I get the following error while trying to create a PDF via the XeLaTeX engine:

> Error producing PDF.
> ! Missing $ inserted.
> <inserted text>
>                 $
> l.116 \caption{foo_bar}

Removing/replacing the underscore fixes the issue.

**Comment by tolot27** [comment]
That's a problem of LaTeX, not pandoc-crossref. Try `foo\_bar` or simply avoid underscores in captions. See also https://tex.stackexchange.com/a/65387/153215

**Comment by lierdakil** [comment]
No, this is a bug alright -- crossref should escape it when necessary.

**Comment by lierdakil** [comment]
Should be fixed in `master`. Can't promise a release soon though.

**Comment by lierdakil** [comment]
Released in 0.3.1.0
